Source/Core/DolphinQt/Host.cpp

```diff
@@ -162,6 +165,17 @@ bool Host::GetGBAFocus()
 #endif
 }
 
+bool Host::GetTASInputFocus() const
+{
+  QWidget* const active_window = QApplication::activeWindow();
```

Comment:
Have you checked that this is fine to call on threads other than the main thread, and that it's fast?

Reply:
I've switched to using an atomic bool which gets set when a TASInputWindow receives an ActivationChange event, so now only the Host thread touches Qt objects.